frontend/public/components/nav/nav-header.tsx

```diff
@@ -1,25 +1,64 @@
 import * as React from 'react';
 import { connect } from 'react-redux';
 import { Dropdown, DropdownItem, DropdownToggle, Title } from '@patternfly/react-core';
-import { CaretDownIcon } from '@patternfly/react-icons';
+import { CaretDownIcon, CubesIcon } from '@patternfly/react-icons';
 import { Perspective, useExtensions, isPerspective } from '@console/plugin-sdk';
 import { RootState } from '../../redux';
 import { featureReducerName, getFlagsObject, FlagsObject } from '../../reducers/features';
-import { history } from '../utils';
-import { useActivePerspective } from '@console/shared';
+import { Firehose, history } from '../utils';
+import { FLAGS, useActivePerspective } from '@console/shared';
+import { K8sResourceKind, referenceForModel } from '../../module/k8s';
+import { ConsoleLinkModel } from '../../models';
+import * as _ from 'lodash-es';
 
 type StateProps = {
   flags: FlagsObject;
 };
 
 export type NavHeaderProps = {
   onPerspectiveSelected: () => void;
+  consoleLinks?: any;
 };
 
-const NavHeader_: React.FC<NavHeaderProps & StateProps> = ({ onPerspectiveSelected, flags }) => {
+export const ACM_LINK_ID = 'acm-console-link';
+
+const NavHeaderWrapper: React.FC<NavHeaderProps & StateProps> = ({
+  onPerspectiveSelected,
+  flags,
+}) => {
+  const resources = [];
+  if (flags[FLAGS.CONSOLE_LINK]) {
+    resources.push({
+      kind: referenceForModel(ConsoleLinkModel),
+      isList: true,
+      prop: 'consoleLinks',
+    });
+  }
+
+  return (
+    <Firehose resources={resources}>
+      <NavHeaderContents onPerspectiveSelected={onPerspectiveSelected} flags={flags} />;
+    </Firehose>
+  );
+};
+
+const getACMConsoleLink = (links): K8sResourceKind => {
```

Comment:
Give the argument a type. (We should probably add a `ConsoleLinkKind` type, but not necessary for this PR.)

```suggestion
const getACMConsoleLink = (links: K8sResourceKind[]): K8sResourceKind => {
```

Reply:
fixed